mobify/platform-scaffold
# `export default customCommands` so importing is more natural

Initial feedback from Debenhams is that using custom commands from the IM feels different than built-in commands. 

## Before

The following example contrasts importing built-in commands and custom commands:
```js
// Built-in
import {login} from 'progressive-web-sdk/dist/integration-manager/account/commands'
dispatch(login('jeremy@mobify.com', 'supersecretpassword', true))

// Custom
import {customCommands} from 'progressive-web-sdk/dist/integration-manager/custom/commands'
return dispatch(customCommands.isEmailAvailableCommand(formValues.username))
```

## After
This PR changes the export of custom commands slightly so that importing them is more like the built-in commands.

```js
// Built-in (unchanged)
import {login} from 'progressive-web-sdk/dist/integration-manager/account/commands'
dispatch(login('jeremy@mobify.com', 'supersecretpassword', true))

// Custom (this works)
import {isEmailAvailableCommand} from 'progressive-web-sdk/dist/integration-manager/custom/commands'
return dispatch(isEmailAvailableCommand(formValues.username))

// Custom (this also works)
import * as customCommands from 'progressive-web-sdk/dist/integration-manager/custom/commands'
return dispatch(customCommands.isEmailAvailableCommand(formValues.username))
```

## Missing Custom Command

If the custom command is _not_ registered, it is still possible to check if the custom command is defined and skip calling it. A bonus is that the console warns that the command is also undefined which is a good signal that the UI is dispatching a custom command that no longer exists.

```js
import {isEmailAvailableCommand} from '../../integration-manager/custom/commands'
if (isEmailAvailableCommand) {
    return dispatch(isEmailAvailableCommand(formValues.username))
}
```

![image](https://user-images.githubusercontent.com/77138/27309915-94838f5a-550b-11e7-955a-8b23f3433a5c.png)

 **JIRA**: N/A
 **Linked PRs**: N/A

## Changes
- Export the `customCommands` as the default export instead of being a named export
- Update the single custom command usage to work with new export model

## Todos

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Add an item to the cart and start checkout
- Enter an email address in the Email Confirmation field and verify that a network call is made to the `isEmailAvailable` endpoint
